frontend/dogplus-frontend/src/components/social/FriendsAndRequests.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useState, useEffect } from "react";
+import { PublicUser } from "../../types/user";
+import { FriendRequest } from "../../types/social";
+import toast from "react-hot-toast";
+
+interface FriendsAndRequestsProps {
+  refreshKey: number;
+}
+
+const FriendsAndRequests = (props: FriendsAndRequestsProps) => {
+  const [activeTab, setActiveTab] = useState("friends");
+  const [friends, setFriends] = useState<PublicUser[]>([]);
+  const [pendingRequests, setPendingRequests] = useState<FriendRequest[]>([]);
+  const [sentRequests, setSentRequests] = useState<FriendRequest[]>([]);
+
+  useEffect(() => {
+    fetchFriends();
+    fetchPendingRequests();
+    fetchSentRequests();
+  }, []);
+
+  useEffect(() => {
+    fetchSentRequests();
+  }, [props.refreshKey]);
+
+  const fetchFriends = async () => {
+    const response = await fetch(
+      `${process.env.REACT_APP_BACKEND_HOST}/api/social/followers/`,
+      {
+        headers: { Authorization: `Token ${localStorage.getItem("token")}` },
+      }
+    );
+    if (response.ok) {
+      const data = await response.json();
+      setFriends(
+        data
+          .filter((f: FriendRequest) => f.is_accepted)
+          .map((f: any) => f.follower)
+      );
+    }
+  };
+
+  const fetchPendingRequests = async () => {
+    const response = await fetch(
+      `${process.env.REACT_APP_BACKEND_HOST}/api/social/follow-requests/`,
+      {
+        headers: { Authorization: `Token ${localStorage.getItem("token")}` },
+      }
+    );
+    if (response.ok) {
+      const data = await response.json();
+      setPendingRequests(data.filter((f: FriendRequest) => !f.is_accepted));
+    }
+  };
+
+  const fetchSentRequests = async () => {
+    const response = await fetch(
+      `${process.env.REACT_APP_BACKEND_HOST}/api/social/sent-requests/`,
+      {
+        headers: { Authorization: `Token ${localStorage.getItem("token")}` },
+      }
+    );
+    if (response.ok) {
+      const data = await response.json();
+      setSentRequests(data);
+    }
+  };
+
+  const cancelRequest = async (requestId: number) => {
+    const response = await fetch(
+      `${process.env.REACT_APP_BACKEND_HOST}/api/social/cancel-request/${requestId}/`,
+      {
+        method: "DELETE",
```

Comment:
After testing on google cloud, the DELETE method does not work. Could be changed to a post?